### PR TITLE
chore(generative-ai): add url endpoints for compass web usage COMPASS-8376

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-8376

These endpoints don't exist yet in Atlas, so they may be subject to change a little. I'll open a follow up pr if so. This pr is focuses on making these urls configurable by the consumer of the Atlas AI service (web vs electron).
Ideally these would use the same endpoint, however, as mentioned in a comment in the code, we cannot do UI calls against the admin API at the moment (CLOUDP-251201). 

Testing wise we have [e2e tests](https://github.com/mongodb-js/compass/blob/2b44d2663f9ab824985d1abdf858b6d7361a835f/packages/compass-e2e-tests/tests/collection-ai-query.test.ts#L88) for the endpoints being used in Compass, and existing unit tests. We'll be adding e2e tests for Compass web gen ai in COMPASS-8395. Also a quick manual test worked.

### CODE REVIEW

File: packages/compass-web/src/entrypoint.tsx [hunk 1]
```diff
@@ -59,11 +62,23 @@ import { useCompassWebLoggerAndTelemetry } from './logger-and-telemetry';
...
       <AtlasServiceProvider>
-        <AtlasAiServiceProvider>{children}</AtlasAiServiceProvider>
+        <AtlasAiServiceProvider
+          apiURLPreset="cloud"
+          urlConfig={{
+            'user-access': (userId: string) =>
+              aiURLConfig.cloud['user-access'](userId, projectId),
+            query: aiURLConfig.cloud.query(projectId),
+            aggregation: aiURLConfig.cloud.aggregation(projectId),
+          }}
```
Comment (gribnoysup) [inline packages/compass-web/src/entrypoint.tsx:78]: I think we touched in tech design on this, you don't really need to expose the actual config here if you're passing the url preset, the `projectId` also doesn't need to be passed to the component, it's an information available for every connection through connection info provider, so the only thing you need to pass here specifically is the preset type, baking as much of this domain specific logic into the ai provider

Comment (Anemy) [reply packages/compass-web/src/entrypoint.tsx]: Sounds good, thanks for calling it out (again sorry), I moved the configuration of URLs into where they're being used more. I was thinking its better for the logic around the endpoint to happen in the actual service and not the provider. How does that look? This way there's a smaller interface also. I ended up passing in the connection info and not moving around the providers. I think both work so we could always switch it out if we find it more useful to pass the connection info ref in instead.

File: packages/compass-web/src/entrypoint.tsx [hunk 2]
```diff
@@ -59,11 +59,15 @@ import { useCompassWebLoggerAndTelemetry } from './logger-and-telemetry';
 import { type TelemetryServiceOptions } from '@mongodb-js/compass-telemetry';
 import { WorkspaceTab as WelcomeWorkspaceTab } from '@mongodb-js/compass-welcome';
 
-const WithAtlasProviders: React.FC = ({ children }) => {
+const WithAtlasProviders: React.FC<{
+  projectId: string;
```
Comment (gribnoysup) [inline packages/compass-web/src/entrypoint.tsx:63]: Unused now, right?

File: packages/compass-generative-ai/src/atlas-ai-service.ts [hunk 3]
```diff
@@ -192,17 +190,85 @@ export function validateAIAggregationResponse(
...
+          )
+        );
+      }
+
+      const atlasMetadata = connectionInfo?.atlasMetadata;
+      if (!atlasMetadata) {
+        throw new Error(
+          "Can't perform generative ai request: atlasMetadata is not available"
+        );
+      }
```
Comment (gribnoysup) [inline packages/compass-generative-ai/src/atlas-ai-service.ts:255]: Small nit: I guess we never expect the requests with the cloud preset to be done when metadata is missing, so maybe this check be moved to the top of the if to make this clearer

**Review by gribnoysup — APPROVED** [review]
Neat!

File: packages/compass-web/src/entrypoint.tsx (hunk 1 above)
Comment (gribnoysup) [reply packages/compass-web/src/entrypoint.tsx]: > I was thinking its better for the logic around the endpoint to happen in the actual service and not the provider. How does that look?

Yeah, makes absolute sense, looks good!

**Pull request merged**